packages/victory-core/src/victory-util/data.js

```diff
@@ -91,6 +92,67 @@ function cleanData(dataset, props) {
   });
 }
 
+// This method will remove data points that fall outside of the desired domain (non-continuous charts only)
+function formatDataFromDomain(dataset, props) {
+  const exists = (val) => val !== undefined;
+
+  // domain for checking data points that are outside of the bounds
+  // symbol to determine chart types with a single y value point
+  // interpolation for determining chart types with continuous data points that shouldn't be altered
+  const { domain, interpolation } = props;
+
+  if (interpolation) return dataset;
+
+  const minDomainX =
+    !domain || !domain.x ? Domain.getMinFromProps(props, "x") : Collection.getMinValue(domain.x);
+  const maxDomainX =
+    !domain || !domain.x ? Domain.getMaxFromProps(props, "x") : Collection.getMaxValue(domain.x);
+  const minDomainY =
+    !domain || !domain.y ? Domain.getMinFromProps(props, "y") : Collection.getMinValue(domain.y);
+  const maxDomainY =
+    !domain || !domain.y ? Domain.getMaxFromProps(props, "y") : Collection.getMaxValue(domain.y);
+
+  if (!exists(minDomainX) && !exists(maxDomainX) && !exists(minDomainY) && !exists(maxDomainY))
+    return dataset;
+
+  // eslint-disable-next-line complexity
+  return dataset.map((datum) => {
+    let { _x, _y, _y0, _y1 } = datum;
+
+    const baseline = _y0;
+    const value = _y || _y1;
```

Comment:
You'll need a better check for `_y` here, because the value could be zero. Also, I'm not certain, but I think you might want to use `_y1` if it exists, and `_y` otherwise

Reply:
Ah that's right.  Rewriting this.